### PR TITLE
Update README

### PR DESCRIPTION
**Pull request by hexonaut**
Updated to V2 and added a pretty picture + more detail.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,31 @@
 # Direct Deposit Module for Maker
 ![Build Status](https://github.com/makerdao/dss-direct-deposit/actions/workflows/.github/workflows/tests.yaml/badge.svg?branch=master)
 
-The Direct Deposit Module interfaces with third party lending protocols to enable a maximum variable borrow rate for selected assets. Maker Governance is able to pick a target variable borrow rate, and the module will automatically deposit/remove liquidity to ensure that target rate is hit.
+The Dai Direct Deposit Module (D3M) is a tool for directly injecting DAI into third party protocols.
 
-## DssDirectDepositAaveDai
+![D3M](https://imgur.com/kiV7g2f.png)
+
+As seen in the image above, external protocols are viewed under the simplified ERC-4626-like interface. Pool adapters are used to convert protocol complexity into simplified concepts of Excess Capaicty + DAI liquidity + DAI outstanding. How DAI is converted between these states is completely protocol-specific and mostly irrelevant to the D3M.
```
Comment (iamchrissmith) [inline README.md:8]:
```suggestion
As seen in the image above, external protocols are viewed under the simplified ERC-4626-like interface. Pool adapters are used to convert protocol complexity into simplified concepts of Excess Capacity + DAI liquidity + DAI outstanding. How DAI is converted between these states is completely protocol-specific and mostly irrelevant to the D3M.
```

File: README.md [hunk 2]
```diff
@@ -1,9 +1,31 @@
...
+
+The D3M is made of 3 components on the Maker side:
+
+### D3MHub
+
+The primary manager contract responsible for collecting all information and determining which action to take (if any). Each D3M instance is regsitered on the Hub using relevant `file(ilk, ...)` admin functions.
+
+A permissionless `exec(ilk)` function exists which will perform all necessary steps to enforce the maximum borrow rate to within the available liquidity and debt ceiling constraints. `exec(ilk)` will need to be called on a somewhat regular basis to keep the system running properly. Upon unwinding interest will automatically be collected, but you can also permissionlessly call `reap(ilk)` to pull in all collected interest as well.
```
Comment (iamchrissmith) [inline README.md:16]:
```suggestion
A permissionless `exec(ilk)` function exists which will perform all necessary steps to update the provided liquidity within the debt ceiling and external protocol constraints. `exec(ilk)` will need to be called on a somewhat regular basis to keep the system running properly. Upon unwinding, interest will automatically be collected, but you can also permissionlessly call `reap(ilk)` to pull in all collected interest as well.
```

File: README.md [hunk 3]
```diff
@@ -1,9 +1,31 @@
...
+
+### D3MPool
+
+A "dumb" adapter which is responsible for depositing or withdrawing DAI from the target protocol. Also contains functions which provide information such as `maxDeposit()` and `maxWithdraw()`. These inform the hub the maximum room to deposit or withdraw respectively. Abstractly we view this `maxWithdraw()` value as the available DAI liquidity that can be immediately withdrawn. Different protocols have different mechanisms for making DAI liquidity available. For example, Aave raises interest rates to encourage more 3rd party deposits (or repays from borrowers). In the case of Maple, there is a function `intendToWithdraw()` which signals to the pool operator that they should begin liquidating some positions to free up DAI.
```
Comment (iamchrissmith) [inline README.md:20]:
```suggestion
A "dumb" adapter which is responsible for depositing or withdrawing DAI from the target protocol based on instructions from the Hub. Also contains functions which provide information such as `maxDeposit()` and `maxWithdraw()`. These inform the hub the maximum room to deposit or withdraw respectively. Abstractly we view this `maxWithdraw()` value as the available DAI liquidity that can be immediately withdrawn. Different protocols have different mechanisms for making DAI liquidity available. For example, some protocols raise interest rates to encourage more 3rd party deposits (or repays from borrowers). In the other cases, there could be a function such as `intendToWithdraw()` which signals to the pool operator that they should begin liquidating some positions to free up DAI.
```

File: README.md [hunk 4]
```diff
@@ -1,9 +1,31 @@
...
+
+### D3MPlan
+
+The D3MPlan can be viewed as the controller for D3M instances. The plan is responsible for reading all relevant information (maybe nothing) from the protocol and condensing this down to a debt target. This desired target debt is forwarded to the Hub to take action to reach this target debt level asap.
```
Comment (iamchrissmith) [inline README.md:24]:
```suggestion
The D3MPlan can be viewed as the targeting logic for D3M instances. The plan is responsible for reading all relevant information from its state (i.e. the target rate) and possibly from the external protocol (i.e. current balance of supply and borrowing in the market) and condensing this down to a debt target. This desired target debt is forwarded to the Hub to take action to reach this target debt level asap.
```

File: README.md [hunk 5]
```diff
@@ -1,9 +1,31 @@
 # Direct Deposit Module for Maker
 ![Build Status](https://github.com/makerdao/dss-direct-deposit/actions/workflows/.github/workflows/tests.yaml/badge.svg?branch=master)
 
-The Direct Deposit Module interfaces with third party lending protocols to enable a maximum variable borrow rate for selected assets. Maker Governance is able to pick a target variable borrow rate, and the module will automatically deposit/remove liquidity to ensure that target rate is hit.
+The Dai Direct Deposit Module (D3M) is a tool for directly injecting DAI into third party protocols.
 
-## DssDirectDepositAaveDai
+![D3M](https://imgur.com/kiV7g2f.png)
```
Comment (iamchrissmith) [inline README.md:6]: I like this image.  You might add somehow that the pool holds any relevant balances (i.e. vat debt and any protocol tokens)

**Review by iamchrissmith — CHANGES_REQUESTED** [review]
This looks good! I made a couple of suggestions to wording and one comment on the image for your consideration

File: README.md (hunk 5 above)
Comment (hexonaut) [reply README.md]: done

**Review by iamchrissmith — APPROVED** [review]
lgtm

**Pull request merged**